scripts/build_airocean_parameters.py

```diff
@@ -0,0 +1,558 @@
+        [
+            [-0.3557814, 0.84358, -0.40223423, 1.0],
+            [-0.41468223, 0.65596241, 0.63067581, 1.0],
+            [-0.58849102, 0.53029673, 0.0627648, 1.0],
+        ],
+        [
+            [-0.99500944, 0.0913478, -0.04014718, 1.0],
+            [-0.58849102, 0.53029673, 0.0627648, 1.0],
+            [-0.41468223, 0.65596241, 0.63067581, 1.0],
+        ],
+    ]
+)
+ico_20_centers = np.array(
+    [
+        [0.6446662, 0.27407261, 0.37518719],
+        [0.17476898, 0.52317601, 0.57203007],
+        [-0.16999525, 0.11746359, 0.76731978],
+        [0.08682596, -0.38238388, 0.69117259],
+        [0.59031442, -0.28559418, 0.44882132],
+        [0.67643404, 0.37526316, -0.18190733],
+        [0.22617043, 0.68690576, -0.32936779],
+        [-0.08387563, 0.77832093, 0.13659114],
+        [-0.64171587, 0.12186443, 0.45257654],
+        [-0.67643404, -0.37526316, 0.18190733],
+        [-0.22617043, -0.68690576, 0.32936779],
+        [0.08387563, -0.77832093, -0.13659114],
+        [0.58849102, -0.53029673, -0.0627648],
+        [0.64171587, -0.12186443, -0.45257654],
+        [-0.59031442, 0.28559418, -0.44882132],
+        [-0.6446662, -0.27407261, -0.37518719],
+        [-0.17476898, -0.52317601, -0.57203007],
+        [0.16999525, -0.11746359, -0.76731978],
+        [-0.58849102, 0.53029673, 0.0627648],
+        [-0.08682596, 0.38238388, -0.69117259],
+    ]
+)
+# Centers corresponding to the split faces of Australia
+ico_2_centers = np.array(
+    [
+        [-0.07609745, 0.53600476, -0.6075312],
+        [-0.09755446, 0.22876301, -0.77481398],
+    ]
+)
+# Centers corresponding to the split faces around Japan
+ico_3_centers = np.array(
+    [
+        [-0.64642729, 0.48840818, -0.12653887],
+        [-0.45298488, 0.67661305, 0.09706879],
+        [-0.66606089, 0.42586898, 0.21776448],
+    ]
+)
+
+"""
+    Definition of the forward and inverse transform matrices from icosahedral faces to airocean unfolded net space
+"""
+
+A = ico_faces[:, 1] - ico_faces[:, 0]
+B = ico_faces[:, 2] - ico_faces[:, 0]
+ico_normals = -np.array(
+    [
+        A[:, 1] * B[:, 2] - A[:, 2] * B[:, 1],
+        A[:, 2] * B[:, 0] - A[:, 0] * B[:, 2],
+        A[:, 0] * B[:, 1] - A[:, 1] * B[:, 0],
+    ]
+).T
+ico_normals /= ((ico_normals**2).sum(axis=1) ** 0.5).reshape(-1, 1)
+ico_centers = ico_faces.mean(axis=1)
+ico_elevated_centers = ico_centers.copy()
+ico_elevated_centers[:, :3] += ico_normals
+
+airocean_basis = np.zeros((23, 4, 4))
+airocean_basis[:, :3, :] = airocean_23_face_vertices[:, :3, :]
+airocean_basis[:, 3, :] = airocean_23_centers
+
+ico_basis = np.zeros((23, 4, 4))
+ico_basis[:, :3, :] = ico_faces
+ico_basis[:, 3, :] = ico_elevated_centers
+
+
+air_ico_trans = np.array(
+    [
+        a.T @ np.linalg.inv(b.T)
+        for i, (a, b) in enumerate(zip(ico_basis, airocean_basis))
+    ]
+)
+ico_air_trans = np.array(
+    [
+        b.T @ np.linalg.inv(a.T)
+        for i, (a, b) in enumerate(zip(ico_basis, airocean_basis))
+    ]
+)
```

Comment:
(missed this one from my previous review). Using the `zap_zero_or_one` function that I mentioned earlier, many near zeros and ones can be set exactly zero and one. These were introduced from the iterative matrix inversion algorithms. I suggest to amend this:
```python
air_ico_trans = zap_zero_or_one(air_ico_trans)
ico_air_trans = zap_zero_or_one(ico_air_trans)
```
with `verbose=True` the messages show many tiny adjustments:
```
adjusted 27 zeros
adjusted 8 ones
adjusted 36 zeros
adjusted 6 ones
```